frontend/src/components/app/app.tsx

```diff
@@ -1,22 +1,47 @@
-import { AppRoute } from 'common/enums/enums';
 import { FC } from 'common/types/types';
-import { Routes, Route } from 'components/common/common';
+import { MainPage } from 'pages/main-page';
 import { Auth } from 'components/auth/auth';
+import { AppRoute } from 'common/enums/enums';
+import { useLocation } from 'react-router-dom';
 import { NotFound } from '../not-found-page/not-found';
+import { Routes, Route } from 'components/common/common';
+import { isRouteHaveHeader } from 'helpers/routes/is-route-have-header';
+import { HeaderContainer } from 'components/common/header/header-container';
+import { SidebarContainer } from 'components/common/sidebar/sidebar-container';
 import { ConfirmationModalTest } from './tests/confirmation-modal/confirmation-modal';
-import { MainPage } from 'pages/main-page';
+
+import styles from './app.module.scss';
 
 const App: FC = () => {
+  const { pathname } = useLocation();
+
   return (
     <>
-      <Routes>
-        <Route path={AppRoute.ROOT} element={<MainPage />} />
-        <Route path={AppRoute.SIGN_UP} element={<Auth />} />
-        <Route path={AppRoute.SIGN_IN} element={<Auth />} />
-        <Route path={AppRoute.RESTORE_PASSWORD} element={<Auth />} />
-        <Route path={AppRoute.ANY} element={<NotFound />} />
-        <Route path={'test/confirmationModal/'} element={<ConfirmationModalTest />} />
-      </Routes>
+      {isRouteHaveHeader(pathname) && (
+        <Routes>
+          <Route path={AppRoute.SIGN_UP} element={<Auth />} />
+          <Route path={AppRoute.SIGN_IN} element={<Auth />} />
+          <Route path={AppRoute.RESTORE_PASSWORD} element={<Auth />} />
+        </Routes>
+      )}
+      {!isRouteHaveHeader(pathname) && (
```

Comment:
There is not reason to call `isRouteHaveHeader` method twice. Move result to separate const